modules/swift/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,310 @@
+= Working with Vector Search
+:page-status: Beta
+:page-edition: Enterprise
+:page-aliases: 
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:description: Use Vector Search with Full Text Search and Query.
+:keywords: edge AI api swift ios macos apple vector search generative
+
+[abstract]
+{description}
+
+== Use Vector Search
+
+To configure a project to use vector search, follow the xref:swift:gs-install.adoc[installation instructions] to add the Vector Search extension.
+
+NOTE: You must install Couchbase Lite to use the Vector Search extension.
+
+== Create a Vector Index
+
+This method shows how you can create a vector index using the Couchbase Lite Vector Search extension.
+
+[source, swift]
+----
+
+func vectorIndex() throws {
+    
+    database = try Database(name: "vectorDB")
+    collection = try database.defaultCollection()
+        
+    // Create a default Vector Index Configuration    
+    var config = VectorIndexConfiguration(expression: "string", dimensions: 300, centroids: 20)
+        
+     // Set custom optional settings
+     config.encoding = .none
+     config.metric = .cosine
+     config.minTrainingSize = 50
+     config.maxTrainingSize = 300
+        
+}
+----
+
+First, initialise the `config` variable with the `VectorIndexConfiguration()` method with the following parameters:
+
+* The data stored within this index will be of type `string`.
+
+* The width or `dimensions` of the vector index is set to `300`.
+
+* The amount of `centroids` is set to `20`.
+This means that there will be 20 buckets with a single centroid each that gathers together similar vectors.
+
+There are also some default values that we did not set with the `VectorIndexConfiguration()` method, these are:
+
+* The encoding type - 8 bit Scalar Quantization.
+
+* The distance metric - Squared Euclidean.
+
+* The minimum and maximum number of vectors for training the index.
+
+NOTE: The number of vectors, the width or dimensions of the vectors and the training size can incur high CPU and memory costs as the size of each variable increases.
+This is because the training vectors have to be resident on the machine.
+
+You can also alter some optional config settings such as `encoding`.
+
+=== Vector Index Configuration
+
+The table below displays the different configurations you can modify within your `VectorIndexConfiguration()` function.
+For more information on specific configurations, see xref:swift:vector-search.adoc[Vector Search.]
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+.Vector Index Configuration Options
+[cols ="4*"]
+|===
+|Configuration Name |Is Required |Default Configuration |Further Information
+
+|Expression
+|Required
+|No default
+| A SQL++ expression indicating where to get the vectors. 
```

Comment:
Check that this renders OK. You may need to use `{sqlpp}` to prevent the + signs from getting mangled by asiidoctor